src/Sulu/Bundle/RouteBundle/Controller/RouteController.php

```diff
@@ -167,20 +167,23 @@ public function cdeleteAction(Request $request)
     private function generateUrlResponse(Request $request)
     {
         $resourceKey = $this->getRequestParameter($request, 'resourceKey');
+        $locale = $this->getRequestParameter($request, 'locale');
         $resourceKeyMapping = $this->resourceKeyMappings[$resourceKey] ?? null;
 
         /** @var array $parts */
         $parts = $this->getRequestParameter($request, 'parts', true);
 
         $route = '/' . \implode('-', $parts);
         if ($resourceKeyMapping) {
+            $options = $this->resourceKeyMappings[$resourceKey]['options'];
+            $options['locale'] = $locale;
             $route = $this->routeGenerator->generate($parts, $this->resourceKeyMappings[$resourceKey]['options']);
```

Comment:
think you need to pass `$options` here? 🤔

Reply:
would be nice to add a functional test case in the `RouteControllerTest`

Reply:
I think we need todo this in the #5965